STUDIO/org.openl.rules.webstudio/src/org/openl/rules/rest/DesignTimeRepositoryService.java

```diff
@@ -203,4 +205,13 @@ public Repository getRepositoryByName(String repoName) {
         }
         throw new NotFoundException("design.repo.message", new String[] { repoName });
     }
+
+    private void allowedToPush(Repository repo) {
+        if (repo.supports().branches()) {
+            BranchRepository branchRepo = (BranchRepository) repo;
+            if (branchRepo.isBranchProtected(branchRepo.getBranch())) {
+                throw new ForbiddenException("default.message");
+            }
+        }
+    }
 }

```

Comment:
Probably we need to do similar changes to RepositoryService because it's another way to modify projected branches.
If nobody uses that class, we should remove it.
But AFAIK, it's still used by some teams.

Reply:
Done